browser/resources/settings/shortcuts_page/components/ConfigureShortcut.tsx

```diff
@@ -56,41 +56,46 @@ const keyTransforms = {
 const getKey = (key: string) => keyTransforms[key] || key
 
 class AcceleratorInfo {
-  codes: string[] = []
-  keys: string[] = []
+  get codes() {
+    return [...this.#modifiers, this.#key?.code!].filter(k => k)
```

Comment:
Curious what is the purpose of last `!` in last `.code!`?

Reply:
Essentially it lets me assert the type:

```ts
type Thing = number
type MaybeThing = Thing | undefined

let f: MaybeThing

let g = f! // type of g is Thing
```

it's not strictly correct here, as code could be null/undefined, but it means Typescript infers the correct type for this array (`string[]` instead of `(string | number)[]`). 

I could also write 

```tsx
return [...this.#modifiers, this.#key?.code].filter(k => k) as string[]
```

Reply:
Ah, it's non-null assertion operator! in TS. I didn't know that. Thanks a lot 👍🏼